streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -257,24 +275,39 @@
                     REPLICATION_FACTOR_DOC)
             .define(TIMESTAMP_EXTRACTOR_CLASS_CONFIG,
                     Type.CLASS,
-                    FailOnInvalidTimestamp.class.getName(),
+                    null,
                     Importance.MEDIUM,
                     TIMESTAMP_EXTRACTOR_CLASS_DOC)
+            .define(DEFAULT_TIMESTAMP_EXTRACTOR_CLASS_CONFIG,
+                        Type.CLASS,
+                        FailOnInvalidTimestamp.class.getName(),
+                        Importance.MEDIUM,
+                        DEFAULT_TIMESTAMP_EXTRACTOR_CLASS_DOC)
             .define(PARTITION_GROUPER_CLASS_CONFIG,
                     Type.CLASS,
                     DefaultPartitionGrouper.class.getName(),
                     Importance.MEDIUM,
                     PARTITION_GROUPER_CLASS_DOC)
             .define(KEY_SERDE_CLASS_CONFIG,
-                    Type.CLASS,
-                    Serdes.ByteArraySerde.class.getName(),
-                    Importance.MEDIUM,
-                    KEY_SERDE_CLASS_DOC)
+                        Type.CLASS,
+                        null,
+                        Importance.MEDIUM,
+                        KEY_SERDE_CLASS_DOC)
+            .define(DEFAULT_KEY_SERDE_CLASS_CONFIG,
+                        Type.CLASS,
+                        Serdes.ByteArraySerde.class.getName(),
+                        Importance.MEDIUM,
+                        DEFAULT_KEY_SERDE_CLASS_DOC)
             .define(VALUE_SERDE_CLASS_CONFIG,
                     Type.CLASS,
-                    Serdes.ByteArraySerde.class.getName(),
+                    null,
```

Comment:
If we want to distinguish between `VALUE_SERDE_CLASS_CONFIG` and `DEFAULT_VALUE_SERDE_CLASS_CONFIG` for example, in     `public Serde valueSerde()`  method, we have to differentiate whether 

` Serde<?> serde = getConfiguredInstance(VALUE_SERDE_CLASS_CONFIG, Serde.class);
`
is default or not. If it is default, then we will call `defaultValueSerde()` method. If we don't set the default value to `null`, then its default value will be some Object initialized with `Serdes.ByteArraySerde.class.getName()` class. In this case, it is hard to know whether the old `serde` (`VALUE_SERDE_CLASS_CONFIG`) is overridden (so we return it) or it has  default value (so we call `defaultValueSerde() `method). 

Moreover, because we are controlling the access to `VALUE_SERDE_CLASS_CONFIG` via `valueSerde()` method, we handle `null` cases here as well.

The same applies to `KEY_SERDE_CLASS_CONFIG` and `TIMESTAMP_EXTRACTOR_CLASS_CONFIG` as well.